packages/flutter_tools/lib/src/resident_web_runner.dart

```diff
@@ -130,33 +151,54 @@ class ResidentWebRunner extends ResidentRunner {
       printError(message);
       return 1;
     }
-    // Start the web compiler and build the assets.
-    await webCompilationProxy.initialize(
+
+    /// Start a build daemon.
+    final DaemonHandle handle = await webCompilationProxy.daemon(
       projectDirectory: FlutterProject.current().directory,
       targets: <String>[target],
+      release: debuggingOptions.buildInfo.isRelease,
     );
-    _lastCompiled = DateTime.now();
-    final AssetBundle assetBundle = AssetBundleFactory.instance.createBundle();
-    final int build = await assetBundle.build();
-    if (build != 0) {
-      throwToolExit('Error: Failed to build asset bundle');
-    }
-    await writeBundle(fs.directory(getAssetBuildDirectory()), assetBundle.entries);
-
-    // Step 2: Start an HTTP server
-    _server = WebAssetServer(flutterProject, target, ipv6);
-    await _server.initialize();
-
-    // Step 3: Spawn an instance of Chrome and direct it to the created server.
-    final String url = 'http://localhost:${_server.port}';
-    final Chrome chrome = await chromeLauncher.launch(url);
-    final ChromeTab chromeTab = await chrome.chromeConnection.getTab((ChromeTab chromeTab) {
-      return chromeTab.url.contains(url); // we don't care about trailing slashes or #
-    });
-    _connection = await chromeTab.connect();
-    _connection.onClose.listen((WipConnection connection) {
-      exit();
-    });
+
+    /// Start the webdev server?
+    final Chrome chrome = await chromeLauncher.launch('???');
```

Comment:
The initialization logic here isn't quite clear yet, the webdev server needs the devtools, which needs a url - from chrome?

Reply:
The `DevTools` class is really just serving devtools itself. It takes only a hostname, which can be whatever the user wants it to be. I think in webdev we choose the hostname that the user chose for webdev itself.

Reply:
Ahh, our devtools, not chrome devtools - understood